pandas/src/period.pyx

```diff
@@ -36,6 +36,8 @@ from tslib cimport (
     _nat_scalar_rules,
 )
 
+from pandas.tseries import frequencies
+
```

Comment:
what you need to do is at the top here type and import all of the freqs that you need e.g

```
cdef int D_RESO = frequencies.D_RESO
...
```